bundle/config/mutator/python/python_mutator.go

```diff
@@ -266,7 +290,23 @@ func writeInputFile(inputPath string, input dyn.Value) error {
 	return os.WriteFile(inputPath, rootConfigJson, 0600)
 }
 
-func loadOutputFile(rootPath string, outputPath string) (dyn.Value, diag.Diagnostics) {
+// loadLocationsFile loads locations.json containing source locations for generated YAML.
+func loadLocationsFile(locationsPath string) (*pythonLocations, error) {
+	if _, err := os.Stat(locationsPath); os.IsNotExist(err) {
+		return newPythonLocations(), nil
+	}
+
+	locationsFile, err := os.Open(locationsPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open locations file: %w", err)
+	}
+
+	defer locationsFile.Close()
```

Comment:
I'm not 100% sure of the exact order. I think it works because defer is executed only on exit. I will rewrite the code in a way which doesn't require such knowledge :)

Reply:
All `defer` statements are executed upon exiting the frame after the return value is computed.

Reply:
https://go.dev/ref/spec#Defer_statements

Reply:
Ok, so this should be fine.